src/wiki/plugins/macros/mdx/toc.py

```diff
@@ -6,16 +8,92 @@
 HEADER_ID_PREFIX = "wiki-toc-"
 
 
+def process_toc_depth(toc_depth):
+    if isinstance(toc_depth, str) and "-" in toc_depth:
+        toc_top, toc_bottom = [int(x) for x in toc_depth.split("-")]
+    else:
+        toc_top = 1
+        toc_bottom = int(toc_depth)
+    return {"toc_top": toc_top, "toc_bottom": toc_bottom}
+
+
+def process_value(org_val, new_val):
+    try:
+        if type(new_val) is str:
+            new_val = new_val.lstrip("'").rstrip("'")
+            if type(org_val) is not str:
+                new_val = json.loads(new_val.lower())
```

Comment:
What is going on here? Maybe I'm forgetting something, I haven't been around this codebase for a long time. But I don't understand how JSON comes into the picture :)

Reply:
I'm a little concerned here that it'd be possible to have `json.loads` called with data from an attacker.

Reply:
Derp! You are totally right. I didn't think of that. 
The reason was I was thinking about the future. Markdown 3.4.4 TOC extension supports an argument 'toc_class' which can take a list of classes to provide the div containing the TOC. I wanted to make sure I could handle all data types. It was a little extra weight and now that you mention it a bad idea.

Instead, I should just assume for now that the value is one of the following, str, int, bool. And act accordingly.

Reply:
Thanks for aligning with the Markdown TOC extension as well :100: